doc/how_to_guides/pilz_industrial_motion_planner/src/pilz_sequence.cpp

```diff
@@ -0,0 +1,366 @@
+
+  // [ --------------------------------------------------------------- ]
+  // [ ------------------ MoveGroupSequence Action ------------------- ]
+  // [ --------------------------------------------------------------- ]
+  // Plans and executes the trajectory
+
+  // MoveGroupSequence action client
+  using MoveGroupSequence = moveit_msgs::action::MoveGroupSequence;
+  auto action_client = rclcpp_action::create_client<MoveGroupSequence>(node, "/sequence_move_group");
+
+  // Verify that the action server is up and running
+  if (!action_client->wait_for_action_server(std::chrono::seconds(10)))
+  {
+    RCLCPP_ERROR(LOGGER, "Error waiting for action server /sequence_move_group");
+    return -1;
+  }
+
+  // Create a MotionSequenceRequest
+  moveit_msgs::msg::MotionSequenceRequest sequence_request;
+  sequence_request.items.push_back(item1);
+  sequence_request.items.push_back(item2);
+
+  // Create action goal
+  auto goal_msg = MoveGroupSequence::Goal();
+  goal_msg.request = sequence_request;
+
+  // Planning options
+  goal_msg.planning_options.planning_scene_diff.is_diff = true;
+  goal_msg.planning_options.planning_scene_diff.robot_state.is_diff = true;
+  // goal_msg.planning_options.plan_only = true;
+
+  // Goal response callback
+  auto send_goal_options = rclcpp_action::Client<MoveGroupSequence>::SendGoalOptions();
+  send_goal_options.goal_response_callback = [](std::shared_ptr<GoalHandleMoveGroupSequence> goal_handle) {
+    try
+    {
+      if (!goal_handle)
+      {
+        RCLCPP_ERROR(LOGGER, "Goal was rejected by server");
+      }
+      else
+      {
+        RCLCPP_INFO(LOGGER, "Goal accepted by server, waiting for result");
+      }
+    }
+    catch (const std::exception& e)
+    {
+      RCLCPP_ERROR(LOGGER, "Exception while waiting for goal response: %s", e.what());
+    }
+  };
+
+  // Result callback
+  send_goal_options.result_callback = [](const GoalHandleMoveGroupSequence::WrappedResult& result) {
+    switch (result.code)
+    {
+      case rclcpp_action::ResultCode::SUCCEEDED:
+        RCLCPP_INFO(LOGGER, "Goal succeeded");
+        break;
+      case rclcpp_action::ResultCode::ABORTED:
+        RCLCPP_ERROR(LOGGER, "Goal was aborted. Status: %d", result.result->response.error_code.val);
+        break;
+      case rclcpp_action::ResultCode::CANCELED:
+        RCLCPP_ERROR(LOGGER, "Goal was canceled");
+        break;
+      default:
+        RCLCPP_ERROR(LOGGER, "Unknown result code");
+        break;
+    }
+    RCLCPP_INFO(LOGGER, "Result received");
+  };
+
+  // Feedback callback
+  // It does not show the state of the sequence, but the state of the robot: PLANNING, MONITOR, IDLE
+  send_goal_options.feedback_callback = [](GoalHandleMoveGroupSequence::SharedPtr,
+                                           const std::shared_ptr<const GoalHandleMoveGroupSequence::Feedback> feedback) {
+    RCLCPP_INFO(LOGGER, "Feedback: %s", feedback->state.c_str());
+  };
```

Comment:
Do we need a feedback callback?

Reply:
Actually, no. I thought the feedbak would return the next point in the sequence during the execution, but since it only returns the state of the action is not really that usefull (at least not in this example)